fuzz/fuzz_targets/deserialize_br.rs

```diff
@@ -17,7 +18,31 @@ fuzz_target!(|data: &[u8]| {
 
     let mut allocator = Allocator::new();
     let program = node_from_bytes_backrefs(&mut allocator, &b1).unwrap();
-
+    let program_old = node_from_bytes_backrefs_old(&mut allocator, &b1).unwrap();
     let b2 = node_to_bytes_backrefs(&allocator, program).unwrap();
     assert_eq!(b1, b2);
+    let b3 = node_to_bytes_backrefs(&allocator, program_old).unwrap();
+    assert_eq!(b1, b3);
+});
+
+#[cfg(feature = "counters")]
+fuzz_target!(|data: &[u8]| {
+    let mut allocator = Allocator::new();
+    let program = match node_from_bytes_backrefs(&mut allocator, data) {
+        Err(_) => {
+            return;
+        }
+        Ok(r) => r,
+    };
+
+    let b1 = node_to_bytes_backrefs(&allocator, program).unwrap();
+
+    // reset allocators
+    let mut allocator = Allocator::new();
+    let mut allocator_old = Allocator::new();
+
+    let mut allocator = Allocator::new();
+    let program = node_from_bytes_backrefs(&mut allocator, &b1).unwrap();
+    let program_old = node_from_bytes_backrefs_old(&mut allocator_old, &b1).unwrap();
+    assert!(allocator.pair_count() <= allocator_old.pair_count());
```

Comment:
Another important test (that I believe would fail right now) is to deserialize `bytes` again, using both old and new functions and an allocator that has a pair-limit of `allocator_old.pair_count() - 1`. i.e. it should run out of memory, and so should the new function. Otherwise consensus rules will have changed, and we have a potential for a chain split.

I believe the simplest way to implement this is to have a function on `Allocator` to create a "fake" pair. I just needs to decrement its limit. It also means `checkpoint()` and `restore_checkpoint()` would need to track and update this limit.

Reply:
I think it would be great to also ensure that the deseialization yields the same tree. You can use `node_eq` (see other fuzzers). This precludes using a single `Allocator` though, as I suggested earlier.